### ISSUE TITLE
Pattern matching on interned values

### ISSUE THREAD

**Issue by Kixiron** [vmware/differential-datalog#821]
Being able to pattern match on interned values (similar to how `Ref<'T>` works) would be a great quality of life tweak

**Comment by ryzhyk** [comment]
Yes, I forget what the exact issue was, but I don't think this is a fundamental limitation.
